File: backend/vision/backend/app/schemas/gesture.py
from pydantic import BaseModel


class GestureEvent(BaseModel):

    type: str

    source: str

    detector: str

    gesture: str

    confidence: float

    timestamp: float

    bbox: list[float] | None = None

    payload: dict | None = None


class GestureAction(BaseModel):

    gesture: str

    action: str

File: frontend/src/services/gesture.ts
export type GestureSource = 'mediapipe' | 'yolo' | 'pointer' | 'keyboard'
export type GestureAction =
  | 'rotate'
  | 'shuffle'
  | 'select'
  | 'confirm'
  | 'reset'
  | 'stop'
  | 'switch_spread'
  | 'request_reading'

export interface GestureEvent {
  source: GestureSource
  gesture: string
  action: GestureAction
  confidence: number
  delta?: number
  timestamp: number
}

export interface YoloGesturePayload {
  gesture: string
  confidence: number
  action: GestureAction
  timestamp?: number
}
